### PR TITLE
[DNM] [android] Updated NDK to 12

### PR DESCRIPTION
**Pull request by alexzatsepin**
@therearesomewhocallmetim @rokuz @syershov @bykoianko

### CODE REVIEW

File: android/jni/Application.mk [hunk 1]
```diff
@@ -8,7 +8,7 @@ APP_STL := c++_static
 APP_CPPFLAGS += -Wno-deprecated-register
 
 ifeq (x$(NDK_ABI_TO_BUILD), x)
-  APP_ABI := armeabi-v7a-hard x86
+  APP_ABI := armeabi-v7a
```
Comment (Dushistov) [inline android/jni/Application.mk]: Why remove `x86` here?

File: android/build.gradle [hunk 2]
```diff
@@ -220,7 +220,7 @@ android {
   splits.abi {
     enable true
     reset()
-    include 'x86', 'armeabi-v7a'
+    include 'armeabi-v7a'
```
Comment (Dushistov) [inline android/build.gradle]: And why here remove x86?

Comment (alexzatsepin) [inline android/build.gradle]: For the same reason. It's temporary. First I wanted to update NDK at least for arm architecture to get it (https://developer.android.com/studio/projects/add-native-code.html ) work

**Comment by Dushistov** [comment]
@alexzatsepin Not get it. I have similar patch, but for NDK 13. It has several additional changes in compare with this pull request and related to diff 12..13, but not many.

And it compiles fine for arm and for x86, also it works in emulator/x86. 

So what you want to achive, ability to debug C++ code in Android Studio?

>Or probably we will remove x86 completely if the arm app works fine on x86 in emulation mode

qemu that emulates arm on x86 is so slow, any news that anybody speedup qemu of the order of magnitude?

**Comment by alexzatsepin** [comment]
> So what you want to achive, ability to debug C++ code in Android Studio?

Exactly. Plus we need to update NDK to newer version anyway.

`qemu that emulates arm on x86 is so slow, any news that anybody speedup qemu of the order of magnitude?`

Removing the x86 is definitely the last action which we will apply. Of course, we should try to keep x86. And the support of emulators is the really important thing, because as you said, the arm emulator is still slow.

I've never heard about 'qemu', if you clarify it for me I will be appreciate.

**Comment by Dushistov** [comment]
@alexzatsepin 

>I've never heard about 'qemu', if you clarify it for me I will be appreciate.

Android emulator based on qemu opensource project: http://wiki.qemu.org/Main_Page
For example you can look at `Android/Sdk/tools/qemu` or run emulator and find "*qemu*" process in `top`.

File: tools/mkspecs/android-g++/qmake.conf [hunk 3]
```diff
@@ -63,7 +63,7 @@ contains(ANDROID_TARGET_ARCH, arm) {
     QMAKE_CFLAGS += -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16
     QMAKE_LFLAGS += -march=armv7-a -Wl,--fix-cortex-a8
   } else {
-    contains(ANDROID_TARGET_ABI, armeabi-v7a-hard) {
```
Comment (Dushistov) [inline tools/mkspecs/android-g++/qmake.conf]: I compare your patch with mine line by line, and
at now it looks strange:
contains(ANDROID_TARGET_ABI, armeabi-v7a) {
...
  } else {
    contains(ANDROID_TARGET_ABI, armeabi-v7a) {
  ... 
   } else {
...

why two times handle `contains(ANDROID_TARGET_ABI, armeabi-v7a) ` ?

**Comment by alexzatsepin** [comment]
Updated to ndk 13 already

**Pull request closed without merging**